controllers/cinder_controller.go

```diff
@@ -872,7 +877,7 @@ func (r *CinderReconciler) generateServiceConfigs(
 
 	labels := labels.GetLabels(instance, labels.GetGroupLabel(cinder.ServiceName), serviceLabels)
 
-	db, err := mariadbv1.GetDatabaseByName(ctx, h, cinder.DatabaseName)
+	db, err := mariadbv1.GetDatabaseByNameAndAccount(ctx, h, instance.Name, instance.Spec.DatabaseAccount, instance.Namespace)
```

Comment:
Why is this necessary?
- Isn't the db supplied as an input (L870)? L880 simply supersedes the input param.
- If the caller isn't supplying the correct 'db' object then it be fixed in the caller's code.

Reply:
sure, the presence of `GetDatabaseByName` here should have been removed as part of #349 which rearranged the flow to pass `db` in.  So I can remove it here and make sure tests are not impacted.